pkg/handler/collector/github/github_test.go

```diff
@@ -0,0 +1,173 @@
+package github
+
+import (
+	"context"
+	"errors"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/guacsec/guac/internal/testing/dochelper"
+	"github.com/guacsec/guac/internal/testing/testdata"
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/guacsec/guac/pkg/logging"
+)
+
+// FIXME: Test is not running properly due to an indexing error
+func Test_github_RetrieveArtifacts(t *testing.T) {
+	con, cancel := context.WithTimeout(context.Background(), time.Second*3)
+	defer cancel()
+	ctx := logging.WithLogger(con)
+	logger := logging.FromContext(ctx)
+
+	docs := []*processor.Document{
+		{
+			Blob:   dochelper.ConsistentJsonBytes(testdata.GitHubAssetExample1),
+			Type:   processor.DocumentUnknown,
+			Format: processor.FormatUnknown,
+			SourceInformation: processor.SourceInformation{
+				Collector: string(CollectorGitHubDocument),
+				Source:    "v1.4.0",
+			},
+		},
+		{
+			Blob:   dochelper.ConsistentJsonBytes(testdata.GitHubAssetExample2),
+			Type:   processor.DocumentUnknown,
+			Format: processor.FormatUnknown,
+			SourceInformation: processor.SourceInformation{
+				Collector: string(CollectorGitHubDocument),
+				Source:    "v1.4.0",
+			},
+		},
+		{
+			Blob:   dochelper.ConsistentJsonBytes(testdata.GitHubAssetExample3),
+			Type:   processor.DocumentUnknown,
+			Format: processor.FormatUnknown,
+			SourceInformation: processor.SourceInformation{
+				Collector: string(CollectorGitHubDocument),
+				Source:    "v1.4.0",
+			},
+		},
+	}
+	type fields struct {
+		poll     bool
+		token    string
+		owner    string
+		repo     string
+		tag      string
+		tagList  []string
+		interval time.Duration
+	}
+	tests := []struct {
+		name                   string
+		fields                 fields
+		numberOfFilesCollected int
+		wantErr                bool
+		errMessage             error
+		want                   []*processor.Document
+	}{{
+		// TODO: Fix fields for test cases below
+		name: "Get all sboms",
+		fields: fields{
+			poll:     false,
+			token:    os.Getenv("API_KEY"),
+			owner:    "slsa-framework",
+			repo:     "slsa-github-generator",
+			tag:      "v1.4.0",
+			interval: time.Millisecond,
+		},
+		numberOfFilesCollected: 3,
+		wantErr:                false,
+		want:                   docs,
+	}, {
+		name: "Tag not specified",
+		fields: fields{
+			poll:     false,
+			token:    os.Getenv("API_KEY"),
+			owner:    "slsa-framework",
+			repo:     "slsa-github-generator",
+			tag:      "",
+			interval: time.Millisecond,
+		},
+		want:    docs,
+		wantErr: false,
+	}, {
+		name: "No repo provided",
+		fields: fields{
+			poll:     false,
+			token:    os.Getenv("API_KEY"),
+			owner:    "",
+			repo:     "",
+			interval: time.Millisecond,
+		},
+		errMessage: errors.New("GET https://api.github.com/repos///releases/latest: 404 Not Found []"),
+		wantErr:    true,
+	}, {
+		name: "Poll latest release",
```

Comment:
Does the polling test case not return anything?

Reply:
Does it make sense to even do the polling test case?

Reply:
Polling is a hard test case to do. Maybe we have it as part of an integration test instead of unit test.

Reply:
hmm, I think I may remove this test case for the time being then.